src/cnaas_nms/db/settings.py

```diff
@@ -683,3 +751,91 @@ def get_group_regex(group_name: str) -> Optional[str]:
             continue
         if group_name == group['group']['name']:
             return group['group']['regex']
+
+
+def get_groups_priorities(hostname: Optional[str] = None,
+                          settings: Optional[dict] = None) -> Dict[str, int]:
+    """Return dicts with {name: priority} for groups"""
+    groups_priorities = {}
+
+    if not settings:
+        settings, _ = get_group_settings()
+    if settings is None:
+        return groups_priorities
+    if 'groups' not in settings:
+        return groups_priorities
+    if settings['groups'] is None:
+        return groups_priorities
+    for group in settings['groups']:
+        if 'name' not in group['group']:
```

Comment:
Seems like you are checking for all sorts of paramaters and skipping when certain keys don't exist? Perhaps it is better to not except the yaml file when parsing it, if there are syntax errors or keys missing? 

It seems to me that the iteration through the groups will be a lot easier.

Something like 

```python
for key, value in settings["group"].items():
    do the logic
```

Reply:
We should check the syntax of the yaml file but I think a lot of the stuff are optional, so there might not be a group defined at all etc